ocs_ci/deployment/deployment.py

```diff
@@ -652,3 +667,131 @@ def create_catalog_source(image=None, ignore_upgrade=False):
     )
     # Wait for catalog source is ready
     catalog_source.wait_for_state("READY")
+
+
+def setup_local_storage():
+    """
+    Setup the necessary resources for enabling local storage.
+
+    """
+    # Get the worker nodes
+    workers = get_typed_nodes(node_type='worker')
+    worker_names = [worker.name for worker in workers]
+    worker_names_str = " ".join(worker_names)
+    logger.debug("Workers: %s", workers)
+
+    # Label the worker nodes
+    logger.info("Labeling worker nodes")
+    run_cmd(
+        f"oc label nodes {worker_names_str} "
+        f"{constants.OPERATOR_NODE_LABEL}"
+    )
+
+    logger.info("Retrieving local-storage-operator data from yaml")
+    lso_data = list(templating.load_yaml(
+        constants.LOCAL_STORAGE_OPERATOR, multi_document=True
+    ))
+    # Update local-storage-operator subscription data with channel
+    for data in lso_data:
+        if data['kind'] == 'Subscription':
+            data['spec']['channel'] = config.ENV_DATA.get('ocs_version')
+
+    # Create temp yaml file and create local storage operator
+    logger.info(
+        "Creating temp yaml file with local-storage-operator data:\n %s",
+        lso_data
+    )
+    lso_data_yaml = tempfile.NamedTemporaryFile(
+        mode='w+', prefix='local_storage_operator', delete=False
+    )
+    templating.dump_data_to_temp_yaml(
+        lso_data, lso_data_yaml.name
+    )
+    logger.info("Creating local-storage-operator")
+    run_cmd(f"oc create -f {lso_data_yaml.name}")
+
+    local_storage_operator = ocp.OCP(
+        kind=constants.POD, namespace='local-storage'
+    )
+    assert local_storage_operator.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=constants.LOCAL_STORAGE_OPERATOR_LABEL,
+        timeout=600
+    ), "Local storage operator did not reach running phase"
+
+    # Retrieve NVME device path ID for each worker node
```

Comment:
I wonder if from this point on is not too specific for i3?
If we can make this method more generic for ALL LSO deployments, it will be great

Reply:
I've moved the device path code to its own function and added a TODO for determining patterns for vmware and bare metal. Once we know how to find the device paths for those platforms I can add that logic in a follow up PR.